app/controllers/groups/samples_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Groups
+  # Controller actions for Samples within a Group
+  class SamplesController < ApplicationController
+    layout 'groups'
+    before_action :context_crumbs, only: %i[index]
+
+    def index
+      namespaces = Namespaces::ProjectNamespace.where(parent_id: group.self_and_descendant_ids)
+      @pagy, @samples = pagy(Sample.where(project_id: Project.where(namespace: namespaces)).includes(:project))
```

Comment:
You should be able to simplify these 2 lines into one:

`@pagy, @samples = pagy(Sample.where(project: Project.where(namespace: Namespaces::ProjectNamespace.where(parent_id: group.self_and_descendant_ids))))`

Reply:
All good! The only benefit is it reduces the number of lines in half to accomplish the same task. But yeah definitely for readability it's better! I'll leave it upto you!